src/Symfony/Component/Form/Form.php

```diff
@@ -304,7 +304,7 @@ public function setData(mixed $modelData): static
         if (!FormUtil::isEmpty($viewData)) {
             $dataClass = $this->config->getDataClass();
 
-            if (null !== $dataClass && !$viewData instanceof $dataClass) {
+            if (null !== $dataClass && !$viewData instanceof $dataClass && $this->name !=='autocomplete') {
```

Comment:
This change appears to be incorrect as the name of the form field is an arbitrary value.